StoreKitUnitTests/StoreProductTests.swift

```diff
@@ -112,4 +106,34 @@ class StoreProductTests: StoreKitConfigTestCase {
         expect(storeProduct.subscriptionGroupIdentifier) == "7096FF06"
     }
 
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func testSk2PriceFormatterFormatsCorrectly() async throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        let productIdentifier = "com.revenuecat.monthly_4.99.1_week_intro"
+        let sk2Fetcher = ProductsFetcherSK2()
+
+        let storeProductSet = try await sk2Fetcher.products(identifiers: Set([productIdentifier]))
+
+        let storeProduct = try XCTUnwrap(storeProductSet.first)
+        let priceFormatter = try XCTUnwrap(storeProduct.priceFormatter)
+        let productPrice = storeProduct.price as NSNumber
+
+        expect(priceFormatter.string(from: productPrice)) == "$4.99"
```

Comment:
_Not for this PR_
It would be nice to have tests that ensure this formats in other locales too. The new Test Plans in Xcode makes that a lot easier.
Otherwise if we changed the implementation to hardcode this to Dollars it would still pass.

Reply:
I _think_ I can do that with `StoreKitTest`, I'll take a look